nntrainer/layers/lstm.cpp

```diff
@@ -74,7 +138,72 @@ void LSTMLayer::setRecurrentActivation(ActivationType activation) {
 }
 
 void LSTMLayer::forwarding(bool training) {
-  // NYI
+  Tensor &weight_xh =
+    weightAt(static_cast<int>(LSTMParams::weight_xh)).getVariableRef();
+  Tensor &weight_hh =
+    weightAt(static_cast<int>(LSTMParams::weight_hh)).getVariableRef();
+  Tensor &bias_h =
+    weightAt(static_cast<int>(LSTMParams::bias_h)).getVariableRef();
+
+  Tensor &hidden_ = net_hidden[0]->getVariableRef();
+  Tensor &input_ = net_input[0]->getVariableRef();
+  Tensor &m_cell_ = mem_cell->getVariableRef();
+
+  Tensor temp;
+  Tensor hs_prev;
+  Tensor cs_prev;
+  Tensor hs;
+  Tensor cs;
+  Tensor f;
+  Tensor g;
+  Tensor i;
+  Tensor o;
+
+  for (unsigned int b = 0; b < input_dim[0].batch(); ++b) {
+    Tensor islice = input_.getBatchSlice(b, 1);
+    Tensor oslice = hidden_.getBatchSlice(b, 1);
+    Tensor cell = m_cell_.getBatchSlice(b, 1);
+
+    for (unsigned int t = 0; t < islice.height(); ++t) {
+      Tensor xs = input_.getSharedDataTensor(TensorDim(1, 1, 1, islice.width()),
+                                             t * islice.width());
```

Comment:
Very minor code clean idea :)
```suggestion
      Tensor xs = input_.getSharedDataTensor({ islice.width() }), t * islice.width());
```